### PR TITLE
3007 improvement see plan  add ability to createedit plan

### LINKED ISSUE #3007: [Improvement] 'See Plan' | add ability to create/edit plan
The issue is that user is not able to add plan to empty day. User can only select Planned dates.

Expect: 
1. When the user selects 'See Plan' and tab 'Future' should be able to select the not planned day, and the button below should become 'Add Plan' instead 'Select' (we keep 'Select' for days with plans)

2. In tab 'Tomorrow' button 'Edit Plan' should be active even if warnings are not resolved

Future
Screenshot at Sep 10 09-49-48

Tomorrow
Screenshot at Sep 10 09-51-02

### PR DESCRIPTION
**Pull request by CREDO23**
## Description

This PR fixes #3007 

## Type of Change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Breaking change
-   [ ] Documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings

## Current screenshots

[Loom](https://www.loom.com/share/8e366e31f01b4f73a941734ac23d0fba?sid=62a52cd7-2f55-4655-9fc2-a6daa66bdb48)

### CODE REVIEW

File: apps/web/lib/features/daily-plan/all-plans-modal.tsx [hunk 1]
```diff
@@ -209,63 +227,9 @@ export const AllPlansModal = memo(function AllPlansModal(props: IAllPlansModal)
 										closeModal={handleCloseModal}
 									/>
 								) : (
-									<>
```
Comment (evereq) [inline apps/web/lib/features/daily-plan/all-plans-modal.tsx:212]: @CREDO23 (CC: @Cedric921 ) why we removed that whole block of code here? What it was doing, why we don't need that anymore?

File: apps/web/lib/features/daily-plan/all-plans-modal.tsx [hunk 2]
```diff
@@ -188,13 +200,19 @@ export const AllPlansModal = memo(function AllPlansModal(props: IAllPlansModal)
...
-										onClick={handleCalendarSelect}
+										onClick={selectedPlan ? handleCalendarSelect : createEmptyPlan}
 									>
-										Select
+										{selectedPlan ? (
+											'Select'
+										) : createDailyPlanLoading ? (
+											<SpinnerLoader variant="light" size={20} />
+										) : (
+											'Add plan'
```
Comment (evereq) [inline apps/web/lib/features/daily-plan/all-plans-modal.tsx:214]: @CREDO23 (CC: @Cedric921 ) we need to make sure all "hardcoded" text actually translated in all codebase (i18n)... Please make sure all PRs have that.

File: apps/web/lib/features/daily-plan/add-task-estimation-hours-modal.tsx [hunk 3]
```diff
@@ -664,13 +678,35 @@ function TaskCard(props: ITaskCardProps) {
 		try {
 			setAddToPlanLoading(true);
 
-			if (plan.id) await addTaskToPlan({ taskId: task.id }, plan.id);
+			console.log(plan);
```
Comment (evereq) [inline apps/web/lib/features/daily-plan/add-task-estimation-hours-modal.tsx:681]: @CREDO23 always review EACH line of code in your PRs to make sure you clean up those etc.

File: apps/web/lib/features/daily-plan/all-plans-modal.tsx (hunk 1 above)
Comment (CREDO23) [reply apps/web/lib/features/daily-plan/all-plans-modal.tsx:212]: It is a duplication, the person who worked on #3028 did it !
I think we should reuse existing code ! 
So, I deleted it, and I am fixing #3028 in another PR to reuse existing code ! @evereq

**Pull request merged**